components/content/mention-cache.ts

```diff
@@ -0,0 +1,119 @@
+import { User } from 'game-jolt-frontend-lib/components/user/user.model';
+import { fuzzysearch } from 'game-jolt-frontend-lib/utils/string';
+import { Route } from 'vue-router/types/router';
+import { ContentDocument } from './content-document';
+
+export type MentionCacheUser = {
+	user: User;
+	source: string;
+	rank: number;
+	match: number;
+};
+
+export class MentionCache {
+	private static _currentRoute: Route | null = null;
+	private static _users: MentionCacheUser[] = [];
+
+	private static _isDifferentRoute(route: Route) {
+		return this._currentRoute === null || this._currentRoute.fullPath !== route.fullPath;
+	}
+
+	private static _bustCache(fromRoute: Route) {
+		this._currentRoute = fromRoute;
+		this._users = [];
+	}
+
+	private static _removeBySource(source: string) {
+		this._users = this._users.filter(i => i.source !== source);
+	}
+
+	private static _addUser(user: User, source: string) {
+		const currentUser = this._users.find(i => i.user.id === user.id);
+		if (currentUser) {
+			currentUser.rank++;
+		} else {
+			this._users.push({ user, rank: 1, source, match: 0 });
+		}
+	}
+
+	public static add(fromRoute: Route, source: string, ...users: User[]) {
+		if (this._isDifferentRoute(fromRoute)) {
```

Comment:
What happens to mention cache we gather in a parent route when we only changed a child route and remained on the same parent?

This forces us to always try and gather all the mentions we need in the leaf route where we display the suggestion component. Not sure I like this.